### PR TITLE
BLD: clean-up sdist creation

### PR DESCRIPTION
**Pull request by TomAugspurger**
I messed up in https://github.com/pandas-dev/pandas/pull/28374/files#diff-2eeaed663bd0d25b7e608891384b7298R531. I meant to revert the lines that excluded cythonized files in the sdist to save it for a followup.

So I'm using this to clean up the sdist logic a bit, and exclude a few test data files that have snuck into our distribution.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
LGTM

**Review by WillAyd — APPROVED** [review]
lgtm as well - merge if complete

File: setup.py [hunk 1]
```diff
@@ -524,12 +524,10 @@ def maybe_cythonize(extensions, *args, **kwargs):
     """
     Render tempita templates before calling cythonize
     """
-    if len(sys.argv) > 1 and "clean" in sys.argv:
+    if "clean" in sys.argv or "sdist" in sys.argv:
```
Comment (jorisvandenbossche) [inline setup.py]: can you update the comment here to include this is to not include cythonized files in the sdist as well?

**Comment by TomAugspurger** [comment]
@jorisvandenbossche done and passing.

**Comment by TomAugspurger** [comment]
Whoops, this should have a release note. 1 sec.

File: doc/source/whatsnew/v1.0.0.rst [hunk 2]
```diff
@@ -231,6 +231,8 @@ Build Changes
 ^^^^^^^^^^^^^
 - Fixed pyqt development dependency issue because of different pyqt package name in conda and PyPI (:issue:`26838`)
 - Added a `pyproject.toml <https://www.python.org/dev/peps/pep-0517/>`_ file (:issue:`20775`)
+- Pandas no longer includes cythonized C files in the source distribution uploaded to PyPI. As long as the version of ``pip``
+  supports reading a ``pyproject.toml``, no changes should be required in the build environment. The does not affect users installing wheels (:issue:`28341`).
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.0.0.rst]: It seems a bit strange to me to put the pyproject / cythonized files note in the "bug fixes" section, I would move it above somewhere in enhancements or changes

Comment (TomAugspurger) [reply doc/source/whatsnew/v1.0.0.rst]: Moved to a top-level section after enhancements. I removed the pyqt note, since it's a dev-only thing.

**Review by jorisvandenbossche — APPROVED** [review]

**Review by WillAyd — APPROVED** [review]
Minor typo fixup directly on GH in case you need to re-pull locally, but lgtm. I say merge on green

**Pull request merged**